src/app/modules/widgets/tree-viewer/new-element/new-element.component.ts

```diff
@@ -0,0 +1,59 @@
+import { Component, ViewChild, Input, ElementRef, AfterViewInit, Output, EventEmitter } from '@angular/core';
+import { TreeNode } from '../tree-node';
+
+export enum ContextType { Parent, Sibling }
+export interface NodeContext { node: TreeNode; type: ContextType; }
+
+export interface NewElementConfig {
+  indent: string;
+  context: NodeContext;
+  validateName: (newName: string) => { valid: boolean, message?: string };
+  createNewElement: (context: NodeContext, newName: string) => boolean;
```

Comment:
Do I really need the context here, or is the using component aware of the (node) context in which this closure is executed?
If the context is needed, I would argue that it will probably be useful in checking valid names, too, right?